src/configuration.h

```diff
@@ -19,17 +19,16 @@
 #include <mutex>
 #include <string>
 
+int main(int, char**);
+
 namespace google {
 
 class MetadataAgentConfiguration {
  public:
   MetadataAgentConfiguration();
-  // Parse the command line.
-  // A zero return value means that parsing succeeded and the program should
-  // proceed.  A positive return value means that parsing failed.  A negative
-  // value means that parsing succeeded, but all of the arguments were handled
-  // within the function and the program should exit with a success exit code.
-  int ParseArguments(int ac, char** av);
+  MetadataAgentConfiguration(std::istream& input);
+  MetadataAgentConfiguration(std::istream&& input)
```

Comment:
I'm not sure what's happening here, why do we need two constructors?

Reply:
@ACEmilG added a comment with an explanation.